Algorithm.Framework/Alphas/AlphaModelPythonWrapper.cs

```diff
@@ -24,10 +24,28 @@ namespace QuantConnect.Algorithm.Framework.Alphas
     /// <summary>
     /// Provides an implementation of <see cref="IAlphaModel"/> that wraps a <see cref="PyObject"/> object
     /// </summary>
-    public class AlphaModelPythonWrapper : IAlphaModel
+    public class AlphaModelPythonWrapper : IAlphaModel, INamedModel
     {
         private readonly dynamic _model;
 
+        public string Name
+        {
+            get
+            {
+                using (Py.GIL())
+                {
+                    // if the model defines a Name property then use that
+                    if (_model.HasAttr("Name"))
```

Comment:
@AlexCatarino -- what if the python model derives from a base class, does this check also check the base class? If not, perhaps I need to write a recursive `HasAttr` check -- thoughts?